### PR TITLE
Add docs on how to debug global generators.

### PR DESCRIPTION
**Pull request by godofredoc**
Global generators require sub-build dependencies to validate them. This PR adds some basic documentation to help debugging.

Bug: https://github.com/flutter/flutter/issues/124572

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: ci/builders/README.md [hunk 1]
```diff
@@ -436,4 +432,60 @@ The migration to use the build definition language depends on the release | vali
...
+
+#### Install CAS utility
+
+CAS client is required to download the sub-build artifacts. To install it in your machine run the following steps:
+
+```
+mkdir $HOME/tools
+```
+
+Create an ensure.txt file with the following content:
```
Comment (zanderso) [inline ci/builders/README.md:453]: I suspect a link to the CIPD web interface will be more straightforward.

File: ci/builders/README.md [hunk 2]
```diff
@@ -436,4 +432,60 @@ The migration to use the build definition language depends on the release | vali
...
+# Add $HOME/tools to path and your ~/.bashrc
+```
+
+#### Gclient engine checkout
+
+Create a gclient checkout following instructions from [Setting up the engine environment](https://github.com/flutter/flutter/wiki/Setting-up-the-Engine-development-environment).
+
+### Download sub-builds to the gclient checkout out folder
+
+CAS sub-build artifacts can be downloaded using information from a LUCI build. Using https://ci.chromium.org/p/flutter/builders/prod/Mac%20mac_ios_engine/2818 as an example the execution details from steps 13 - 17 show the commands to download the archives. `-dir` parameter needs to be updated to point to the relative or full path to the out folder in your gclient checkout.
```
Comment (zanderso) [inline ci/builders/README.md:470]: long line. There are others below as well.

File: ci/builders/README.md [hunk 3]
```diff
@@ -436,4 +432,60 @@ The migration to use the build definition language depends on the release | vali
...
+
+These are the commands to execute for our example build:   
+
+```
+pushd <gclient checkout>/src/out
```
Comment (zanderso) [inline ci/builders/README.md:475]: Can any of this be scripted? We can put scripts in the `ci/` directory of the `flutter/engine` repo.

Comment (godofredoc) [reply ci/builders/README.md:475]: It is possible but may require some research. Opened https://github.com/flutter/flutter/issues/124685 to track the CLI potential features.

File: ci/builders/README.md (hunk 2 above)
Comment (godofredoc) [reply ci/builders/README.md:470]: Fixed.

**Review by zanderso — APPROVED** [review]

**Pull request merged**